src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -101,6 +101,8 @@ def normalize_inputs(request: InferRequestBase, inputs: dict) -> dict:
             new_inputs[key] = value
         # If value object has __array__ attribute, load it to Tensor using np.array.
         elif hasattr(value, "__array__"):
+            if hasattr(value, "detach"):
+                value = value.detach().cpu()
```

Comment:
may we have one test for this please?

Reply:
sure, but I need to discuss with python api developers before, is it allowed to use torch in tests or not

Reply:
BTW, @rkazants maybe you know, is there some tricky cases for tf tensors too?

Reply:
Input and output tensors from TF1 can be treated as simple numpy arrays. As for TF2 there is no problem with input tensors but I see for handling output tensors we use `tf_res.numpy()` [here](https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/common/tf2_layer_test_class.py#L63)

Reply:
I think there should not be a problem to use `torch` model only for tests now. We need to update `openvino\src\bindings\python\requirements_test.txt`

Reply:
To be honest I think adding whole PyTorch dependency might be "too heavy" as we are adding only one test to runtime. Are there any other goals? I look at something like ONNX dependency that test many scenarios for ONNX Frontend.

Other than that, why `detach()` is going to be used? It will trigger for all PyTorch based tensors. Is it always a case? As far as I know Tensor created after call to `detach()` [is sharing memory](https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html#torch.Tensor.detach):

> Returned Tensor shares the same storage with the original one. In-place modifications on either of them will be seen, and may trigger errors in correctness checks.

So in specific cases, users are required to secure/not modify original Tensors and to keep alive original Tensors - so memory is not cleared by Python's garbage collector before OV inference. I think this scenarios should be checked first and **I am strongly against "non-copy" approach when dealing with memory that is created outside OpenVINO control flow**. `openvino.runtime.Tensor` class is always copying memory and memory-sharing with numpy arrays is available after explicit flag setting. Please also look at the approach of PyTorch:
![image](https://user-images.githubusercontent.com/65762745/196132391-c9ddd6a9-98d1-485f-856c-d0a42c5ef540.png)

To sum things up, I do not think this change should include `detach()` without copying of memory. It might be dangerous and should be tested out.